File: algorithm/java/prob501-1000/prob563_easy_Binary-Tree-Tilt.java
import java.lang.Math;

class Solution {
    int ans = 0;

    public int findTilt(TreeNode root) {
        if(root == null)
            return 0;
        dfs(root);
        return ans;
    }

    public int dfs(TreeNode node) {
        int left_sum = 0;
        int right_sum = 0;
        if(node.left != null) {
            left_sum = dfs(node.left);
        }
        if(node.right != null) {
            right_sum = dfs(node.right);
        }
        ans += Math.abs(left_sum - right_sum);
        return left_sum + right_sum + node.val;
    }
}

File: algorithm/python/prob501-1000/prob563_easy_Binary-Tree-Tilt.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def findTilt(self, root: TreeNode) -> int:
        if root is None:
            return 0
        self.ans = 0
        self.dfs(root)
        return self.ans

    def dfs(self, node: TreeNode) -> int:
        left_sum = right_sum = 0
        if node.left is not None:
            left_sum = self.dfs(node.left)
        if node.right is not None:
            right_sum = self.dfs(node.right)
        self.ans += abs(left_sum - right_sum)
        return left_sum + right_sum + node.val
